
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
        <meta name="description" content="Documentation for PeakCompare">
      
      
        <meta name="author" content="Sam Fletcher">
      
      
      
        <link rel="prev" href="../peak_calling/">
      
      
      
      <link rel="icon" href="../logo.svg">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.5.48">
    
    
      
        <title>Peak Comparing - PeakCompare Documentation</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.6f8fc17f.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="slate" data-md-color-primary="black" data-md-color-accent="green">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#peak-comparing" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="PeakCompare Documentation" class="md-header__button md-logo" aria-label="PeakCompare Documentation" data-md-component="logo">
      
  <img src="../logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            PeakCompare Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Peak Comparing
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/sof202/PeakCompare" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="PeakCompare Documentation" class="md-nav__button md-logo" aria-label="PeakCompare Documentation" data-md-component="logo">
      
  <img src="../logo.svg" alt="logo">

    </a>
    PeakCompare Documentation
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sof202/PeakCompare" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.7.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Welcome to the PeakCompare wiki
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../metric_interpretation/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Metric interpretation
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../peak_calling/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Peak Calling
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Peak Comparing
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Peak Comparing
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#running" class="md-nav__link">
    <span class="md-ellipsis">
      Running
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Running">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ignoring-this-script" class="md-nav__link">
    <span class="md-ellipsis">
      Ignoring this script
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Ignoring this script">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#grep" class="md-nav__link">
    <span class="md-ellipsis">
      grep
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-the-metric-is-calculated" class="md-nav__link">
    <span class="md-ellipsis">
      How the metric is calculated
    </span>
  </a>
  
    <nav class="md-nav" aria-label="How the metric is calculated">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#pseudopeak-definition" class="md-nav__link">
    <span class="md-ellipsis">
      Pseudopeak definition
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#confidence-interval-calculation" class="md-nav__link">
    <span class="md-ellipsis">
      Confidence interval calculation
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Parameters
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#cutoff" class="md-nav__link">
    <span class="md-ellipsis">
      Cutoff
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#significance" class="md-nav__link">
    <span class="md-ellipsis">
      Significance
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#window-size" class="md-nav__link">
    <span class="md-ellipsis">
      Window size
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#unmerged" class="md-nav__link">
    <span class="md-ellipsis">
      Unmerged
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#time" class="md-nav__link">
    <span class="md-ellipsis">
      Time
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#running" class="md-nav__link">
    <span class="md-ellipsis">
      Running
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Running">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#ignoring-this-script" class="md-nav__link">
    <span class="md-ellipsis">
      Ignoring this script
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Ignoring this script">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#grep" class="md-nav__link">
    <span class="md-ellipsis">
      grep
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-the-metric-is-calculated" class="md-nav__link">
    <span class="md-ellipsis">
      How the metric is calculated
    </span>
  </a>
  
    <nav class="md-nav" aria-label="How the metric is calculated">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#pseudopeak-definition" class="md-nav__link">
    <span class="md-ellipsis">
      Pseudopeak definition
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#confidence-interval-calculation" class="md-nav__link">
    <span class="md-ellipsis">
      Confidence interval calculation
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Parameters
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#cutoff" class="md-nav__link">
    <span class="md-ellipsis">
      Cutoff
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#significance" class="md-nav__link">
    <span class="md-ellipsis">
      Significance
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#window-size" class="md-nav__link">
    <span class="md-ellipsis">
      Window size
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#unmerged" class="md-nav__link">
    <span class="md-ellipsis">
      Unmerged
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#time" class="md-nav__link">
    <span class="md-ellipsis">
      Time
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="peak-comparing">Peak Comparing</h1>
<p>After completing the <a href="../peak_calling/">peak calling pipeline</a> for both your
reference dataset and your comparison dataset, you can now compare regions
between the two. The purpose of this script is to generate a metric that aims
to tell the user how likely that a comparable peak in the comparison dataset
exists (given a peak exists in the reference dataset). Details on how to
interpret the output metric can be found <a href="../metric_interpretation/">here</a>.</p>
<h2 id="running">Running</h2>
<p>Before running this script, it is assumed you have ran the peak calling 
pipeline for both of your datasets. After accomplishing this, identify a peak
(or selection of close by peaks) in the reference dataset (noting down the 
chromosome, start and end base positions). Feed this information into the
configuration file <code>example_config_compare.txt</code>. This configuration file asks
for a few different parameters, head over to [this section] to understand what
these parameters mean and how they are used.</p>
<p>After filling out the configuration file, submit the job to a SLURM queue with</p>
<pre><code class="language-bash">sbatch -p queue .../PeakCompare.sh path/to/config_file.txt
</code></pre>
<h3 id="ignoring-this-script">Ignoring this script</h3>
<p>This script is only given to provide a means for the user to submit the job
into a SLURM queue. You could alternatively just run the underlying python
script yourself. To do this, activate the conda environment with:</p>
<pre><code class="language-bash">conda activate PeakCompare-MACS
</code></pre>
<p>Then run the python script called <code>peak_compare.py</code> which can be found in the
<code>Python_Scripts</code> directory of this repository. There are quite a few arguments
that this script requires due to the multitude of files it needs to read in
(another reason behind the wrapper script). For this reason, you may decide to
create your own wrapper script that achieves your own goals (perhaps you want
to look at multiple regions in the genome simultaneously).</p>
<h4 id="grep">grep</h4>
<p>To save time on reading in files, the wrapper script uses grep to pull out the
chromosome of interest before passing them to <code>peak_compare.py</code>. If you plan on
running the python script interactively (or you are writing your own wrapper
script), consider implementing this as well.</p>
<h2 id="how-the-metric-is-calculated">How the metric is calculated</h2>
<p>The metric is a simple ratio of the number of bases in psuedopeaks in the
comparison and the number of bases in peaks in the reference dataset.
Pseudopeaks is a term that describes regions in your comparison set that are
'almost' peaks when compared to the reference dataset. A more formal definition
is given <a href="#pseudopeak-definition">here</a>.</p>
<p>As alluded to in <a href="../peak_calling/#cutoff-analysis">cutoff analysis</a>, cutoffs
are tricky little things. No matter what cutoff you choose, you will almost
always be classing some regions as peaks because they only just exceed the
cutoff (and classing some regions not as peaks for the opposite reason). This
is problematic when comparing two datasets. How can we determine if a peak is
truly absent in the comparison dataset when there is a possibility that a
single extra read in the region could tip the balance? </p>
<p>The solution given with this tool is to look at the confidence intervals for
each p-value instead of looking at whether they hit certain thresholds. If the
confidence intervals for each dataset overlap, it suggests that (under the
chosen <a href="#significance">significance</a>) the true value could be shared by each
dataset at this point. Considering these intervals is both more lax and strict
than simply using a single threshold value. If the region in the reference
dataset is a very significant peak (very low p-value), then this method
required the same region in the comparison dataset to have similarly small
p-values to be considered as a pseudopeak.</p>
<p>Comparing confidence intervals is useful, but comes with a drawback in this
particular case of using MACS to determine peaks. As mentioned in 
<a href="../peak_calling/#why-merged-and-unmerged-peaks">this section</a>, some peaks
may have a very high p-value (not significant), and yet be called a peak due to
the merging process. This is why both files are required by this python script
and allows us to now formally state what defines a psuedopeak.</p>
<h3 id="pseudopeak-definition">Pseudopeak definition</h3>
<p>Def<sup>n</sup>: A base position is defined to be in a pseudopeak region if
it satisfies either of these criteria:</p>
<p>(i) The base position is within a peak region in the unmerged reference dataset
peaks, and the confidence interval in the comparison dataset overlaps or
exceeds the reference dataset's confidence interval (for this base position).</p>
<p>(ii) The base position surpasses the cutoff threshold in the comparison dataset
and is only within a peak region in the merged reference dataset peaks (not
present in the unmerged peaks).</p>
<h3 id="confidence-interval-calculation">Confidence interval calculation</h3>
<p>It should be noted first that p-values from MACS are calculated using the
Poisson distribution (as pileup counts generally follow this distribution).
Now, the confidence interval calculation is carried out in two steps:</p>
<p>1) A confidence interval is calculated for each position in the bias track.
This requires the following steps:
  - The local variance (see <a href="#window-size">window size</a>) for the bias track is
  calculated.
  - The local standard error is calculated.
  - The confidence interval is found by using <span class="arithmatex">\(\hat\lambda = \lambda \pm
  z_\alpha \cdot std\_err\)</span>. Where <span class="arithmatex">\(z_\alpha\)</span> is the critical value for the
  given <a href="#significance">significance</a>.
2) A confidence interval for the p-value is calculated for each position using
the Poisson distribution with arguments:
  - The <span class="arithmatex">\(\hat\lambda\)</span> confidence interval calculated in 1)
  - The number of reads in that position seen in the coverage (pileup) track</p>
<h2 id="parameters">Parameters</h2>
<p>The underlying python script requires a few parameters that can be specified
in the configuration file. These are:</p>
<ul>
<li><a href="#cutoff">The cutoff</a></li>
<li><a href="#significance">The significance</a></li>
<li><a href="#window-size">The window size</a></li>
<li><a href="#unmerged">'unmerged'</a></li>
</ul>
<h3 id="cutoff">Cutoff</h3>
<p>This is the cutoff that was used with the reference dataset when peak calling.
If you didn't set this value yourself (or simply forgot), you can find out
what cutoff value was used by looking at the log file for the job you
submitted. This should look like:</p>
<pre><code>───────┬───────────────────────────────────
       │ File: 01-Jan~00-12345678.log
───────┼───────────────────────────────────
   1   │ Using a cutoff value of 5.
───────┴───────────────────────────────────
</code></pre>
<h3 id="significance">Significance</h3>
<p>This is the significance used when calculating the confidence intervals
described <a href="#how-the-metric-is-calculated">here</a>. Picking a value that is
smaller will result in tighter confidence intervals and therefore results in
stricter <a href="#pseudopeak-definition">criteria for psuedopeaks</a>. There is no
recommended value here, just be aware that confidence intervals are being
compared against each other (so a significance of 95% is more like 90%).</p>
<h3 id="window-size">Window size</h3>
<p>In calculating the confidence intervals used to determine pseudopeaks, a window
is used to calculate local variance in the bias track. A suitable value for
the window size will be between a quarter and half of the fragment length for
each dataset. It should be noted that a smaller window size will result in the
variance calculated being more localised. However, this comes with the caveat
that less variance can be observed (consider the case when window size is 0). A
careful balance should be chosen here.</p>
<h3 id="unmerged">Unmerged</h3>
<p>You may wish to discount peak regions in the reference dataset that are a
result of the merging process entirely when calculating the metric. To do this
you can specify <code>UNMERGED=1</code> in the configuration file (or use <code>--unmerged</code> if
using the python script). This will mean that the metric will only look at the
ratio of psuedopeaks (in the comparison dataset) with the unmerged peaks (in
the reference dataset). All peaks that are a result of merging will be ignored.</p>
<h2 id="time">Time</h2>
<p>Running this script is fast (a couple of seconds). The main slowdown comes with
reading in files. This is why <a href="#grep">grep</a> is utilised in the provided wrapper
script. The complexity of your data can also increase read times as bedgraph
files (output of most MACS commands) can massively vary in size due to this
factor. The bedgraph file at minimum can be 23 lines long (one for each
chromosome) up to ~2,700,000,000 lines long (one for each base pair).</p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      © 2024, Sam Fletcher
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.6ce7567c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.83f73b43.min.js"></script>
      
        <script src="../javascripts/mathjax.js"></script>
      
        <script src="https://unpkg.com/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>